fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -2132,26 +2142,31 @@ Version getRecoverVersionUnicast(const std::vector<Reference<LogSet>>& logServer
 				populateBitset(versionAllTLogs[k], unknownCommittedVersion.tLogLocIds);
 			}
 		}
-		tLogIdx++;
+	}
+
+	if (versionAllTLogs.empty()) {
+		return std::make_tuple(maxKCV, maxKCV);
 	}
 
 	// Compute recovery version.
+	//
 	// @note we think that the unicast recovery version should always greater than or
-	// equal to "min(DV)" (= "minDV"). To be conservative we use "min(KCV)" (= "minKCV")
+	// equal to "min(DV)" (= "minDV"). To be conservative we use "max(KCV)" (= "maxKCV")
 	// as the default (starting) recovery version and later verify that the computed
 	// recovery version is greater than or equal to "minDV".
-	// @todo modify code to use "minDV" as the default (starting) recovery version
-	// @todo to be investigated: maybe we should use max(KCV) as the starting recovery
-	// version (this is because "known committed version" can advance even after a log
-	// server is locked when unicast is enabled and so all log servers may not preserve
-	// all committed versions, from min(KCV) till the correct recovery version, in
-	// "unknownCommittedVersions" causing the recovery algorithm to not find the correct
-	// recovery version)?
-	Version RV = minKCV; // recovery version
-	std::vector<Version> RVs(maxTLogLocId + 1, minKCV); // recovery versions of various tLogs
-	Version prevVersion = minKCV;
+	//
+	// @note we are not using "min(KCV)" as the default recovery version because "known
+	// committed version" can advance even after a log server is locked when unicast is
+	// enabled and so all log servers may not preserve all committed versions, from
```

Comment:
>  "known committed version" can advance even after a log server is locked when unicast is enabled

in the PR description, can you give an example of how this could happen?

Reply:
Done.